packages/monaco/src/browser/monaco-editor-model.ts

```diff
@@ -124,8 +125,28 @@ export class MonacoEditorModel implements ITextEditorModel, TextEditorDocument {
         return this.model.getVersionId();
     }
 
-    getText(): string {
-        return this.model.getValue();
+    /**
+     * Return selected text by Range or all text by default
+     */
+    getText(range?: Range): string {
+        if (!range) { return this.model.getValue(); }
+        let text: string = '';
```

Comment:
`return this.model.getValueInRange(this.p2m.asRange(range));`

Reply:
Done. That is a cleaner solution.